pkg/manager/manager.go

```diff
@@ -316,6 +319,12 @@ type Runnable interface {
 	Start(context.Context) error
 }
 
+// runnable with preWatch resources
+type RunnableWithPreWatch interface {
```

Comment:
I don't really understand why this would be needed. We already have the `LeaderElectionRunnable` interace to start thigns before the leader lock was acquired: https://github.com/kubernetes-sigs/controller-runtime/blob/8ad090e0663e046c5a9dad68a572af0e30626c85/pkg/manager/manager.go#L330

Moreover, we start the cache before we start anything else: https://github.com/kubernetes-sigs/controller-runtime/blob/8ad090e0663e046c5a9dad68a572af0e30626c85/pkg/manager/internal.go#L466

We don't star the Source itself which might lead to us not getting an informer which then might mean we don't actually start a watch. But changing that is probably a breaking change, as it will lead to increased memory consumption on non-leader instances.

Reply:
@alvaroaleman  thanks for the discussion。 

it has the obvious problem that  it will lead to increased memory consumption on non-leader instances, so i provider one option for users who need this feature.  

in our env(big clusters) and when leader changes, it takes too long time to list resources from apiserver, so we want to save failover time by memory consumption.